### PR TITLE
Prepare 3.13-compatible release

### PR DESCRIPTION
**Pull request by youknowone**
Would it be ok? @lucas42 @cdce8p 

distutils is not included because its test does not pass on 3.13

### CODE REVIEW

**Review by lucas42 — APPROVED** [review]
Looks good to me!
Do you do the actual release process manually?  I've not noticed any code in the repo for releasing.

**Comment by youknowone** [comment]
I built packages with update.py and manaully uploaded it previously. 
I am going to write a github actions file for release this time.

**Comment by youknowone** [comment]
Let's target the release tomorrow.

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      # Used to upload release artifacts
+      contents: write
+      # Used to generate artifact attestation
+      attestations: write
+    steps:
+      - name: Download all the dists
+        uses: actions/download-artifact@v4
+        with:
+          name: python-package-distributions
+          path: dist/
```
Comment (cdce8p) [inline .github/workflows/release.yaml:75]: Not sure this is going to work. Where is `python-package-distributions` defined? You'll probably need something like this:

```suggestion
        with:
          pattern: dist-*
          path: dist
          merge-multiple: true
```

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    permissions:
+      contents: write  # IMPORTANT: mandatory for making GitHub Releases
+      id-token: write  # IMPORTANT: mandatory for sigstore
+
+    steps:
+    - name: Download all the dists
+      uses: actions/download-artifact@v4
+      with:
+        name: python-package-distributions
+        path: dist/
```
Comment (cdce8p) [inline .github/workflows/release.yaml:97]:
```suggestion
```suggestion
        with:
          pattern: dist-*
          path: dist
          merge-multiple: true
```
```

File: .github/workflows/release.yaml (hunk 1 above)
Comment (youknowone) [reply .github/workflows/release.yaml:75]: Thank you, I forgot to change upload part. But I found I cannot upload multiple times to same name anymore.

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.version }}
+          check-latest: true
+      - name: Run tests
+        run: |
+          cd ${{ matrix.folder }}
+          python -m build
```
Comment (cdce8p) [inline .github/workflows/release.yaml:51]:
```suggestion
          cd ${{ matrix.folder }}
          pip install build
          python -m build
```

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -0,0 +1,124 @@
+name: Tests
+
+on:
+  push:
+    tags:
+      - '*'
+  pull_request:  # FIXME: temporary until release.yaml works
```
Comment (cdce8p) [inline .github/workflows/release.yaml:7]:
```suggestion
```

**Review by cdce8p — APPROVED** [review]
Looks good overall! I did a quite more extensive test on my fork and the release was created successfully with all artifacts uploaded.

If you have configured Trusted Publisher in PyPI, I'd say go for it!

File: .github/workflows/release.yaml (hunk 4 above)
Comment (youknowone) [reply .github/workflows/release.yaml]: And thank you so much. This is my first time to setup PyPI release by workflow. It must had taken a lot more time without your help.

**Comment by youknowone** [comment]
The release action: https://github.com/youknowone/python-deadlib/actions/runs/11597133339

**Comment by cdce8p** [comment]
> The release action: https://github.com/youknowone/python-deadlib/actions/runs/11597133339

Saw it just know. Unfortunately there is a small issue with `standard-chunk`. It wasn't uploaded to PyPI. Think I know why as well. Will open a PR shortly.

**Comment by youknowone** [comment]
I am not sure but I messed up something. it was actually released in https://github.com/youknowone/python-deadlib/actions/runs/11596984318/job/32289460524

And yeah, chunk doesn't have tests 😂

**Comment by cdce8p** [comment]
> And yeah, chunk doesn't have tests 😂

Would have been too easy if it worked the first time 😄 Opened #38.

**Comment by lucas42** [comment]
@youknowone Yay, that's an amazing milestone! 🎉 

It's been great working with you on this up until now.  But I don't think I've much to contribute after this point.  I'm not an expert in what any of these libraries are doing internally.
In fact, my main motivation this whole time was to be able to do this pull request on another library, which I'm using in a personal project: https://github.com/beetbox/audioread/pull/145

I wish you all the best in future endeavours!  🙇

**Comment by cdce8p** [comment]
I'm in the same category as @lucas42 here, unfortunately. While I'm able to help out from time to time, my main intention was to get "a" working backport lib going so I can update another project for Python 3.13.

**Comment by youknowone** [comment]
Thank you for the replies. Yes, I will try to keep them up for a while, though I am not a kind of Python code writer anymore 😂
It was all the good experience with you. I also hope I don't need to publish a new version of them for a while. Probably the next release will be related to a new PEP deprecating more libraries.

**Pull request merged**